### PR TITLE
[BE] add readme for .github, located in .github/scripts

### PR DESCRIPTION
**Pull request by clee2000**
Add a README for .github, located in .github/scripts because putting it in .github overwrites the repo README.  

Content is copied from https://github.com/pytorch/pytorch/pull/72943/files, with a few modifications.

### CODE REVIEW

File: .github/scripts/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+repo README.
+
+This directory contains workflows and scripts to support our CI infrastructure that runs on Github Actions.
+
+## Workflows
+
+- Pull CI (`pull.yml`) are run on PRs and on master.
+- Trunk CI (`trunk.yml`) are run on trunk to validate incoming commits. They are usually more expensive to run so we do not run them on PRs unless specified.
+- Scheduled CI (`periodic.yml`) is a subset of trunk CI that is run every few hours on master.
+- Binary CI is run to package binaries for distribution for all platforms.
```
Comment (janeyx99) [inline .github/scripts/README.md:13]:
```suggestion
- Pull CI (`pull.yml`) is run on PRs and on master.
- Trunk CI (`trunk.yml`) is run on trunk to validate incoming commits. Trunk jobs are usually more expensive to run so we do not run them on PRs unless specified.
- Scheduled CI (`periodic.yml`) is a subset of trunk CI that is run every few hours on master.
- Binary CI is run to package binaries for distribution for all platforms.
```

File: .github/scripts/README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+.github/regenerate.sh
+```
+
+### Adding a new generated workflow
+
+New generated workflows can be added in the `.github/scripts/generate_ci_workflows.py` script. You can reference
+examples from that script in order to add the workflow to the stream that is relevant to what you particularly
+care about.
+
+Different parameters can be used to acheive different goals, i.e. running jobs on a cron, running only on trunk, etc.
```
Comment (janeyx99) [inline .github/scripts/README.md:39]: This section is getting phased out (we want to enable people to just modify the workflow files themselves when adding a workflow). Maybe just exclude this section for this PR and add the section of adding a workflow later.

**Comment by janeyx99** [comment]
Added some review. When you're done addressing the comments, feel free to move it out of draft and remember to add a conclusive PR description. Also, this is very much BE, so feel free to add [BE] in the PR title

File: .github/scripts/README.md (hunk 2 above)
Comment (seemethere) [reply .github/scripts/README.md]: Not necessarily since binary builds will still be subject to this

**Review by seemethere — APPROVED** [review]

**Review by janeyx99 — APPROVED** [review]

**Pull request closed without merging**